modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -1119,9 +1130,8 @@ public CompletableFuture<Void> onUpdate(WatchEvent evt) {
                                     .map(NodeWithAttributes::node)
                                     .collect(toSet());
 
-                    newLogicalTopology.forEach(n -> nodesAttributes.put(n.nodeId(), n.nodeAttributes()));
+                    updateLogicalTopologyNodeAttributesAndSaveToVault(newLogicalTopology);
```

Comment:
So, as a result we may finish with an inconsistent vault state, if we will fail right after `updateLogicalTopologyNodeAttributesAndSaveToVault` when applied revision will be x but logicalTopology and node attributes will match x + 1. It's not a problem right?

Reply:
Thanks for noticing! IMHO right behaviour here is to call `updateLogicalTopologyNodeAttributesAndSaveToVault` right after the WatchListener was successfully handled. In that case, your case won't occur, so data in vault will be consistent with `WatchListener`.